pkg/check/table_structure.go

```diff
@@ -50,25 +94,60 @@ func (c *TablesChecker) Check(ctx context.Context) *Result {
 		}
 
 		for _, table := range tables {
+			tableName := dbutil.TableName(schema, table)
 			statement, err := dbutil.GetCreateTableSQL(ctx, c.db, schema, table)
 			if err != nil {
 				markCheckError(r, err)
 				return r
 			}
 
-			err = c.checkCreateSQL(statement)
-			if err != nil {
-				markCheckError(r, errors.Errorf("\n[schema %s, table %s]\n%v", schema, table, err))
+			opts := c.checkCreateSQL(statement)
+			if len(opts) > 0 {
+				options[tableName] = c.checkCreateSQL(statement)
+				statements[tableName] = statement
```

Comment:
do not need `c.checkCreateSQL(statement)` again?

Reply:
no, just record it and display them

Reply:
I mean
```
opts := c.checkCreateSQL(statement)
if len(opts) > 0 {
	options[tableName] = opts
	statements[tableName] = statement
```